File: Release/WetiMatlabFunctions/PostProcessing/PlotTimeResults.m
function PlotTimeResults(TimeResults,PostProcessingConfig)

%% BasicTimePlot
if isfield(PostProcessingConfig.Plots,'BasicTimePlot')
    nFigures        = length(PostProcessingConfig.Plots.BasicTimePlot);    
    for iFigure=1:nFigures
        PlotConfig  = PostProcessingConfig.Plots.BasicTimePlot{iFigure};
        if PlotConfig.Enable
            figure('Name',['BasicTimePlot ',num2str(iFigure)])
            nChannels               = size(PlotConfig.Channels,1);
            % determine IndicesConsideredDataFiles and nConsideredDataFiles
            if isfield(PlotConfig,'IndicesConsideredDataFiles') % plot only considered files
                IndicesConsideredDataFiles  = PlotConfig.IndicesConsideredDataFiles;
                nConsideredDataFiles        = length(IndicesConsideredDataFiles);
            else
                nConsideredDataFiles        = size(TimeResults,1);
                IndicesConsideredDataFiles  = 1:nConsideredDataFiles;                
            end
            % make one subplot for each channel
            for iChannel = 1:nChannels
                ThisChannel     = PlotConfig.Channels{iChannel};
                subplot(nChannels,1,iChannel)                    
                hold on; box on; grid on                
                % plot data
                for iConsideredDataFile=1:nConsideredDataFiles
                    iDataFile               = IndicesConsideredDataFiles(iConsideredDataFile);
                    plot(TimeResults{iDataFile}.(ThisChannel))                                                                                
                end
                % ylabel: use unit from last considered file
                ylabel(['[',TimeResults{iDataFile}.(ThisChannel).DataInfo.Units,']'])
                % plot title: fixed to channels
                title(ThisChannel,'Interpreter','none')
                % plot legend
                if isfield(PlotConfig,'legend')
                    legend(PlotConfig.legend,'Interpreter','none')
                end                
                % set gca properties
                if isfield(PlotConfig,'gca')
                    PropertyFields      = fieldnames(PlotConfig.gca);
                    for iProperty=1:size(PropertyFields,1)
                        Property    = getfield(PlotConfig.gca,PropertyFields{iProperty});
                        set(gca,PropertyFields{iProperty},Property);
                    end
                end
            end
            % xlabel: fixed to time
            xlabel('time [s]')
        end
    end
end

%% ComparisonTimePlot
if isfield(PostProcessingConfig.Plots,'ComparisonTimePlot')
    nFigures     = length(PostProcessingConfig.Plots.ComparisonTimePlot);    
    for iFigure=1:nFigures
        PlotConfig      = PostProcessingConfig.Plots.ComparisonTimePlot{iFigure};
        if PlotConfig.Enable
            figure('Name',['ComparisonTimePlot ',num2str(iFigure)])
            nChannels        = size(PlotConfig.Channels,1);
            % determine IndicesConsideredDataFiles and nConsideredDataFiles
            if isfield(PlotConfig,'IndicesConsideredDataFiles') % plot only considered files
                IndicesConsideredDataFiles  = PlotConfig.IndicesConsideredDataFiles;
                nConsideredDataFiles        = length(IndicesConsideredDataFiles);
            else
                nConsideredDataFiles        = size(TimeResults,1);
                IndicesConsideredDataFiles  = 1:nConsideredDataFiles;                
            end
            % make one subplot for each considered data file       
            for iConsideredDataFile=1:nConsideredDataFiles                    
                subplot(nConsideredDataFiles,1,iConsideredDataFile)
                hold on; box on; grid on                
                iDataFile   = IndicesConsideredDataFiles(iConsideredDataFile);
                for iChannel = 1:nChannels
                    ThisChannel  = PlotConfig.Channels{iChannel};
                    plot(TimeResults{iConsideredDataFile}.(ThisChannel))  
                end
                % ylabel: need to be defined due to different channels 
                if isfield(PlotConfig,'ylabel')
                    ylabel(PlotConfig.ylabel)
                end
                % plot title
                if isfield(PlotConfig,'title')
                    title(PlotConfig.title{iDataFile},'Interpreter','none')
                end
                % plot legend: fixed to Channels
                legend(PlotConfig.Channels,'Interpreter','none')              
                % set gca properties
                if isfield(PlotConfig,'gca')
                    PropertyFields      = fieldnames(PlotConfig.gca);
                    for iProperty=1:size(PropertyFields,1)
                        Property    = getfield(PlotConfig.gca,PropertyFields{iProperty});
                        set(gca,PropertyFields{iProperty},Property);
                    end
                end             
            end 
            % xlabel: fixed to time
            xlabel('time [s]')
        end
    end
end

%% Other more advanced time plots
end

File: Release/PythonFunctions/PostProcessing/PlotTimeResults.py
import matplotlib.pyplot as plt

def PlotTimeResults(time_results, post_processing_config):
    if 'BasicTimePlot' in post_processing_config['Plots']:
        n_figures = len(post_processing_config['Plots']['BasicTimePlot'])
        for i_figure in range(n_figures):
            plot_config = post_processing_config['Plots']['BasicTimePlot'][i_figure]
            if plot_config['Enable']:
                plt.figure(f'BasicTimePlot {i_figure+1}')
                n_channels = len(plot_config['Channels'])
                # determine IndicesConsideredDataFiles and nConsideredDataFiles
                if 'IndicesConsideredDataFiles' in plot_config:  # plot only considered files
                    indices_considered_data_files = plot_config['IndicesConsideredDataFiles']
                    n_considered_data_files = len(indices_considered_data_files)
                else:
                    n_considered_data_files = len(time_results)
                    indices_considered_data_files = list(range(n_considered_data_files))
                # make one subplot for each channel
                for i_channel in range(n_channels):
                    this_channel = plot_config['Channels'][i_channel]
                    this_channel_unit = f"{this_channel}_Unit"
                    plt.subplot(n_channels, 1, i_channel+1)
                    plt.grid(True)
                    # plot data
                    for i_considered_data_file in range(n_considered_data_files):
                        i_data_file = indices_considered_data_files[i_considered_data_file]-1
                        plt.plot(time_results[i_data_file][this_channel])
                    # ylabel: use unit from last considered file
                    plt.ylabel(f"[{time_results[i_data_file][this_channel_unit][0].replace('(', '').replace(')', '')}]")
                    # plot title: fixed to channels
                    plt.title(this_channel)
                    # plot legend
                    if 'legend' in plot_config:
                        plt.legend(plot_config['legend'])
                    # set gca properties
                    if 'gca' in plot_config:
                        for property_field, property_value in plot_config['gca'].items():
                            plt.gca().set(**{property_field: property_value})
                # xlabel: fixed to time
                plt.xlabel('time [s]')
                plt.tight_layout()
    plt.show()

    if 'ComparisonTimePlot' in post_processing_config['Plots']:
        n_figures = len(post_processing_config['Plots']['ComparisonTimePlot'])
        for i_figure in range(n_figures):
            plot_config = post_processing_config['Plots']['ComparisonTimePlot'][i_figure]
            if plot_config['Enable']:
                plt.figure(f'ComparisonTimePlot {i_figure + 1}')
                n_channels = len(plot_config['Channels'])
                # determine IndicesConsideredDataFiles and nConsideredDataFiles
                if 'IndicesConsideredDataFiles' in plot_config:  # plot only considered files
                    indices_considered_data_files = plot_config['IndicesConsideredDataFiles']
                    n_considered_data_files = len(indices_considered_data_files)
                else:
                    n_considered_data_files = len(time_results)
                    indices_considered_data_files = list(range(n_considered_data_files))
                # make one subplot for each considered data file
                for i_considered_data_file in range(n_considered_data_files):
                    plt.subplot(n_considered_data_files, 1, i_considered_data_file + 1)
                    plt.grid(True)
                    i_data_file = indices_considered_data_files[i_considered_data_file]
                    for i_channel in range(n_channels):
                        this_channel = plot_config['Channels'][i_channel]
                        plt.plot(time_results[i_data_file][this_channel])
                    # ylabel: need to be defined due to different channels
                    if 'ylabel' in plot_config:
                        plt.ylabel(plot_config['ylabel'])
                    # plot title
                    if 'title' in plot_config:
                        plt.title(plot_config['title'][i_data_file])
                    # plot legend: fixed to Channels
                    plt.legend(plot_config['Channels'])
                    # set gca properties
                    if 'gca' in plot_config:
                        for property_field, property_value in plot_config['gca'].items():
                            plt.gca().set(property_field, property_value)
                # xlabel: fixed to time
                plt.xlabel('time [s]')
                plt.show()